lib/affinitymap.h

```diff
@@ -0,0 +1,85 @@
+/* Route map function.
+ * Copyright (C) 2022 6WIND
+ *
+ * This file is part of GNU Zebra.
+ *
+ * GNU Zebra is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2, or (at your option) any
+ * later version.
+ *
+ * GNU Zebra is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#ifndef _ZEBRA_AFFINITYMAP_H
+#define _ZEBRA_AFFINITYMAP_H
+
+#include "typesafe.h"
+#include "prefix.h"
+#include "memory.h"
+#include "qobj.h"
+#include "vty.h"
+#include "lib/plist.h"
+#include "lib/plist_int.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define AFFINITY_NAME_SIZE 32
+
+struct affinity_map {
+	char name[AFFINITY_NAME_SIZE];
+	uint16_t bit_position;
+
+	QOBJ_FIELDS;
+};
+DECLARE_QOBJ_TYPE(affinity_map);
+
+struct affinity_maps {
+	struct list *maps;
+
+	bool (*check_use_hook)(const char *affmap_name);
+	void (*update_hook)(const char *affmap_name, uint16_t old_pos, uint16_t new_pos);
+
+	QOBJ_FIELDS;
+};
+DECLARE_QOBJ_TYPE(affinity_maps);
+
+extern const struct frr_yang_module_info frr_affinity_map_info;
+
+int lib_affinity_map_create(struct nb_cb_create_args *args);
+int lib_affinity_map_destroy(struct nb_cb_destroy_args *args);
+int lib_affinity_map_value_modify(struct nb_cb_modify_args *args);
+int lib_affinity_map_value_destroy(struct nb_cb_destroy_args *args);
```

Comment:
I think you are using these as Northbound handlers while initialising the CLI module for affinity maps. If that is correct understanding, you don't need to expose the prototypes here. This is a public include file and should not have these handlers exposed.
All the above 4 prototypes should be removed from this include file. The corresponding function should be made static functions in the same file where the corresponding 'frr_yang_module_info' is defined.

Reply:
done